### PR TITLE
feat: like, bookmark, follow

### PR DESCRIPTION
**Pull request by himadri2110**
### Features:

- Users can like/dislike a post
- Users can add/remove a post to/from Bookmarks. Posts added to bookmarks will show up on the Bookmarks page.
- Users can follow/unfollow other users (list of followers and following will be shown on Profile later)

### CODE REVIEW

**Review by lomesshh — APPROVED** [review]
clean and modular code
UI looks good
like, dislike, bookmark and follow, unfollow all are working as expected
loader looks pretty cool 
looks good to merge

File: src/features/post/components/NewPost.jsx [hunk 1]
```diff
@@ -30,12 +30,19 @@ export const NewPost = ({ post, setShowOptions }) => {
     if (post) newPostRef.current.innerText = post.content;
   }, [post]);
 
+  const focusInput = () => {
+    newPostRef.current && newPostRef.current.focus();
+  };
```
Comment (poornima-alugubelly) [inline src/features/post/components/NewPost.jsx:35]: nice use of ref to achieve focus  👍

File: src/features/post/components/SinglePost.jsx [hunk 2]
```diff
@@ -1,20 +1,32 @@
...
+  const { bookmarks } = useSelector((state) => state.user);
+  const dispatch = useDispatch();
 
   const [showOptions, setShowOptions] = useState(false);
+  const postRef = useRef();
 
   const currentPost = posts.find((post) => post.id === postId);
 
+  const likedByLoggedUser = currentPost?.likes.likedBy.find(
+    (likeUser) => likeUser.username === user.username
```
Comment (poornima-alugubelly) [inline src/features/post/components/SinglePost.jsx]: you can extract this function in such a way that the function where the user is present in a given array , then u can use it in multiple places such as to check like , follow etc

**Review by poornima-alugubelly — APPROVED** [review]
- All features are working as expected
- Good naming of variables

File: src/features/post/components/SinglePost.jsx (hunk 2 above)
Comment (himadri2110) [reply src/features/post/components/SinglePost.jsx]: Extracted the function for now (will update it in the future if needed), thank you.

Comment (himadri2110) [reply src/features/post/components/SinglePost.jsx]: Done, thank you.

Comment (himadri2110) [reply src/features/post/components/SinglePost.jsx]: Done, thank you.

**Pull request merged**